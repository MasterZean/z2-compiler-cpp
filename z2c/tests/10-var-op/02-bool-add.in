{ val a = Bool{1} + Bool{2} }
{ val a = Bool{1} + Small{2} }
{ val a = Bool{1} + Byte{2} }
{ val a = Bool{1} + Short{2} }
{ val a = Bool{1} + Word{2} }
{ val a = Bool{1} + Int{2} }
{ val a = Bool{1} + DWord{2} }
{ val a = Bool{1} + Long{2} }
{ val a = Bool{1} + QWord{2} }
{ val a = Bool{1} + Float{2} }
{ val a = Bool{1} + Double{2} }
{ val a = Bool{1} + Char{2} }
{ val a = Bool{1} + PtrSize{2} }
