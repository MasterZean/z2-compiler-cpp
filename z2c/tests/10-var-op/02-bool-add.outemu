{
	var a: Small = Small{3}
}
{
	var a: Small = Small{3}
}
{
	var a: Byte = Byte{3}
}
{
	var a: Short = Short{3}
}
{
	var a: Word = Word{3}
}
{
	var a: Int = Int{3}
}
{
	var a: DWord = DWord{3}
}
{
	var a: Long = Long{3}
}
{
	var a: QWord = QWord{3}
}
{
	var a: Float = Float{3}
}
{
	var a: Double = Double{3}
}
{
	var a: Char = Char{3}
}
{
	var a: PtrSize = PtrSize{3}
}
