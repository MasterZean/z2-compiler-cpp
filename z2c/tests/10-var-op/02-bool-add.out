{
	int8 a = (bool)(1) + (bool)(2);
}
{
	int8 a = (bool)(1) + (int8)(2);
}
{
	uint8 a = (bool)(1) + (uint8)(2);
}
{
	int16 a = (bool)(1) + (int16)(2);
}
{
	uint16 a = (bool)(1) + (uint16)(2);
}
{
	int32 a = (bool)(1) + (int32)(2);
}
{
	uint32 a = (bool)(1) + (uint32)(2);
}
{
	int64 a = (bool)(1) + (int64)(2);
}
{
	uint64 a = (bool)(1) + (uint64)(2);
}
{
	float a = (bool)(1) + (float)(2);
}
{
	double a = (bool)(1) + (double)(2);
}
{
	uint32 a = (bool)(1) + (uint32)(2);
}
{
	size_t a = (bool)(1) + (size_t)(2);
}
